inst/tests/tests.Rraw

```diff
@@ -19303,12 +19303,6 @@ test(2290.3, DT[, `:=`(a, c := 3)], error="It looks like you re-used `:=` in arg
 # partially-named `:=`(...) --> different branch, same error
 test(2290.4, DT[, `:=`(a = 2, c := 3)], error="It looks like you re-used `:=` in argument 2")
 
-# segfault when selfref is not ok before set #6410
-df = data.frame(a=1:3)
-setDT(df)
-attr(df, "att") = 1
-test(2291.1, set(df, NULL, "new", "new"), error="attributes .* have been reassigned")
-
```

Comment:
So this will only bite us on R < `4.3.0` and will be detected earlier on the ALTREP version?

Reply:
By setting `GROWABLE_BIT` on R ≥ 3.4.0 for `data.table` lists, we prevent the situation where a shallow duplicate of a `data.table` has a misleading `TRUELENGTH`. When duplicating a _growable_ list, R will notice the bit and reset `TRUELENGTH` to 0. This causes `set` to fail an earlier check for non-zero `TRUELENGTH` instead of failing the `.internal.selfref` check, so the error is different ("This data.table has either been loaded from disk...").

Perhaps the test should account for that, especially if we're moving from R-3.3 and thus will no longer hit the old behaviour (where "attributes .* have been reassigned" is still reachable).